shared/styles/index.js.flow

```diff
@@ -48,7 +48,16 @@ declare export var statusBarHeight: number
 declare export var hairlineWidth: number
 declare export function backgroundURL(...path: Array<string>): string
 declare export function styleSheetCreate(obj: Object): Object
-declare export function collapseStyles(styles: any): any
+
+type _Elem = Object | null | false | void
+// CollapsableStyle is a generic version of ?StylesMobile and family,
+// slightly extended to support "isFoo && myStyle".
+export type CollapsibleStyle = _Elem | $ReadOnlyArray<_Elem>
```

Comment:
Why we need this: see e.g. `Avatar`, whose style has type `StylesCrossPlatform`. We then call
```
collapseStyles([props.style, desktopStyles.clickable])
```
so the first element of the array passed to `collapseStyles` may itself be an array, since `StylesCrossPlatform` allows that.

Reply:
I'm a little confused about this. Is the eventual return type of `collapseStyles` going to be `StylesCrossPlatform`?

Reply:
Yeah. Can't do that now because some `style` fields just take an `Object`.

Reply:
those might as well take any at this point if theyre not typed correctly yet. Most are typed Object but its not really the type